src/pydata_sphinx_theme/__init__.py

```diff
@@ -1084,6 +1085,49 @@ def setup_translators(app):
             app.set_translator(name, translator, override=True)
 
 
+# ------------------------------------------------------------------------------
+# customize method for logo management
+# ------------------------------------------------------------------------------
+
+
+def setup_logo_path(
+    app: Sphinx, pagename: str, templatename: str, context: dict, doctree: nodes.Node
+) -> None:
+    """Set up relative paths to logos (works for all Sphinx versions)."""
+
+    # get information from the context "logo_url" for sphinx>=6, "logo" sphinx<6
+    pathto = context.get("pathto")
+    logo = context.get("logo_url") or context.get("logo")
+
+    # logo image is copied to static if set from html_logo in version <6. "_static"
+    # is in the path and in older sphinx versions. to be compatible with all versions,
+    # we remove it manually.
+    logo = Path(logo).name if logo else ""
+
+    # if theme_logo is not set, create a default one
+    if "theme_logo" not in context:
+        context["theme_logo"] = {}
+
+    # reset the "logo" variable for each page
+    context["theme_logo"]["logo"] = {"light": "", "dark": ""}
+
+    # get user-specified light/dark logo versions (fallback to None)
+    image_light = context["theme_logo"].get("image_light")
+    image_dark = context["theme_logo"].get("image_dark")
+
+    # resolve URLs / paths
+    if logo and not isurl(logo):
+        logo = pathto(f"_static/{logo}", resource=True)
+    if image_dark and not isurl(image_dark):
+        image_dark = pathto(f"_static/{image_dark}", resource=True)
+    if image_light and not isurl(image_light):
+        image_light = pathto(f"_static/{image_light}", resource=True)
+
+    # fallback to logo if image_{light|dark} is None
+    context["theme_logo"]["logo"]["light"] = image_light or logo
+    context["theme_logo"]["logo"]["dark"] = image_dark or image_light or logo
```

Comment:
Is this line correct? Our docs say

> If you only specify one of the light or dark variants, the un-specified variant will fall back to the value of html_logo.

So I would have expected 

```suggestion
    context["theme_logo"]["logo"]["dark"] = image_dark or logo
```

...so I'm just checking in case it was a copy-paste error.